DIRHASH_STANDARD.md

```diff
@@ -0,0 +1,203 @@
+Name  | Type             | Default             | Description 
+----  | ---------------- | ------------------- | -----------
+entry_properties | Array of Strings | `["name", "data"]` | Which Directory Entry properties to consider. NOTE that `type` is a mandatory property and should not be provided
+on_cyclic_link | String, One of `"raise"` and `"hash reference"` | `"raise"` | how to handle [cyclic links](#cyclic-links).
+
+The Dirhash Protocol is designed so that the same hash should not be obtained with different Protocol Options. Subsequently, when the same hash is obtained one can be sure that the same Protocol Options were used. The options must still be provided when comparing checksums, but this removes the risk of false positives (confirmation of the checksum) due to wrong options used.
+
+
+
+## Error Conditions
+**Directory Not Accessible**: A not accessible (sub)directory results in an error unless excluded by the `match_patterns` filtering option.
+
+**File Not Accessible**: A not accessible file results in an error if the entry property `data` is used unless excluded by the `match_patterns` filtering option.
+
+**Cyclic Symbolic Links**: Presence of cyclic links, with `on_cyclic_links` set to `raise`.
+
+**Directory Empty**: No (non-empty) directory entries to hash in the Dirhash root directory, given provided Filtering Options and `include_empty_dirs` set to `false`.  
+
+
+
+## The `DIRSUM` Object
+Checksums based on the `DIRSHASH` must contain the additional configuration options to be properly validated (as was discussed in the [Introduction](#introduction)). For this purpose, the Dirhash Standard provides the `DIRSUM` object which contains all the necessary information for verification. It's structure is laid out in JSON below, with properties according [Hash Function](#hash-function-options), [Filtering](#filtering-options) and [Protocol Options](#protocol-options).
+```json
+{
+   "dirhash": "...",
+   "algorithm": "...",
+   "filtering_options": {
+      "match_patterns": ["*"],
+      "linked_dirs": true,
+      "linked_files": true,
+      "empty_dirs": false},
+   "protocol_options": {
+      "entry_properties": ["name", "data"],
+      "on_cyclic_link": "raise"}
+}
+```
+When saved to file, the recommended extension is `<name>.dirsum.json`.
+
+
+## Extensions
+The Dirhash Standard can be extended by introducing additional Filtering Options or entry properties such as:
+- **Permission Properties**: These are hard to get platform independent (different on windows/unix). The best option is probably to let go of platform independence here. Possible properties could be `permission-[owner|group|user]` or `permission-me` for permissions of current 
+process, 
+- **Owning User/Group**
+- **Last Modified/Opened Properties**
+
+
+## Contribute
+If you find a bug, inconsistency or weakness in the Dirhash Standard, or that the documentation or the Standard itself can be simplified without loss of generality, please file an issue!
```

Comment:
put reference to this document here, in case this README gets spread with the wind